pkgs/development/python-modules/qubes-core-qrexec/default.nix

```diff
@@ -0,0 +1,140 @@
+{
+  stdenv,
+  buildPythonPackage,
+  fetchFromGitHub,
+  qubes-core-vchan-xen,
+  pkg-config,
+  pandoc,
+  pyinotify,
+  python,
+  pygobject3,
+  gtk3,
+  wrapGAppsHook3
+}:
+
+let
+  version = "4.2.22";
+  src = fetchFromGitHub {
+    owner = "QubesOS";
+    repo = "qubes-core-qrexec";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-GxT2KpxIOeRF9j9Lc9SRHqM2G58juFtTd/lXoEJzIyY=";
+  };
+
+  agent = stdenv.mkDerivation {
+    inherit version src;
+    pname = "qubes-core-qrexec-agent";
+    sourceRoot = "${src.name}/agent";
+
+    nativeBuildInputs = [
+      pkg-config
+      pandoc
+    ];
+
+    buildInputs = [
+      qubes-core-vchan-xen
+      util-agent-bootstrap
+    ];
+
+    postInstall = ''
+      mv $out/usr/* $out/
+      rm -rf $out/usr
+      # Scripts: qubes-rpc-multiplexer
+      cp ../lib/* $out/bin/
+    '';
+
+    makeFlags = [ "DESTDIR=${placeholder "out"}" ];
+  };
+  # Util hardcodes agent paths, but it is used by agent.
+  # Building separate package for agent, which doesn't use those paths.
+  util-hack = agent: stdenv.mkDerivation {
+    inherit version src;
+    pname = "qubes-core-qrexec-util";
+    sourceRoot = "${src.name}/libqrexec";
+
+    # TODO: qrexec-policy-exec is located in qubes-core-qrexec, thus can't be unhardcoded this easy...
+    # Maybe all of those packages should be built as one, but it is complicated.
+    # As a temporary solution, I think persistent name should be good enough?..
+    patchPhase = ''
+      substituteInPlace qrexec.h \
+        --replace-fail "/var/run/qubes/qrexec-agent-fdpass" "${agent}/bin/qrexec-agent-fdpass" \
+        --replace-fail "/var/run/qubes/qrexec-agent" "${agent}/bin/qrexec-agent" \
+        --replace-fail "/usr/lib/qubes/qubes-rpc-multiplexer" "${agent}/bin/qubes-rpc-multiplexer" \
+        --replace-fail "/usr/bin/qrexec-policy-exec" "/run/current-system/sw/bin/qrexec-policy-exec"
```

Comment:
Definitely not a good idea to use `current-system`. Could we just use `$out` here?

Reply:
This is util package, and qrexec-policy-exec is located in python package, which depends on this one, thus forming a recursive dependency. I have some ideas on how to fix that.

Reply:
I don't think there is a good solution here, those packages are coupled too tightly, and we might want to depend on system environment instead.

We have too many runtime dependencies to resolve here, I think it all needs to be built as a part of one big derivation.

```mermaid
graph LR
  daemon --> util
  agent --> util
  python --> util
  
  util -- qrexec-policy-exec --> python
  util -- qrexec-agent-fdpass, qrexec-agent, qubes-rpc-multiplexer --> agent
  python -- qrexec-client-vm, qubes-rpc-multiplexer --> agent
  python -- qrexec-client --> daemon
```

Reply:
Cycles are resolved